gateway-js/src/__tests__/executeQueryPlan.test.ts

```diff
@@ -164,6 +164,138 @@ describe('executeQueryPlan', () => {
       expect(response).not.toHaveProperty('errors.0.extensions.variables');
     });
 
+    it(`error paths in joins`, async () => {
```

Comment:
good callout. my assumption that i can swap out the `@` with indices from the subgraph doesn't hold. it's easy to get a response path like `["foo", "@", "bar", "@"]` so i need to keep track of the earlier indices somewhere.

Reply:
found a pretty intense solution by examining the router code.

Reply:
Make sense. The solution is indeed a tad involved, but it's not that complicated either and we'd have to refactor quite a bit probably to avoid the "re-hydratation" on errors. Besides, in terms of efficiency, it probably make sense to do that "hydratation" only on errors, so that's fined by me.

Reply:
@pcmanus I added some non-nullable fields in the existing test, and also added a separate test to try out returning the `Query` type. Then I rebased and cleaned everything up. I think we're good to go!